std/datetime/stopwatch.d

```diff
@@ -82,6 +82,22 @@ struct StopWatch
 {
 public:
 
+    /// Measure a time in milliseconds, microseconds, or nanoseconds
+    @safe nothrow @nogc unittest
+    {
+        auto sw = StopWatch(AutoStart.no);
+        sw.start();
+        // ... Insert operations to be timed here ...
+        sw.stop();
```

Comment:
The super-idiomatic use is:

```d
{
    sw.start;
    scope(exit) sw.stop;

    // ... Insert operations to be timed here ...
}
```

but even I use this hardly ;-)
We could add a method that stops when the dtor is called:

```d
with (sw.startScope) {
  // benchmark here
}
```

But I'm not sure whether it's a good idea and worth bothering. https://github.com/dlang/phobos/pull/5416 (dontOptimizeAway) is a lot more important.

Reply:
I actually rather doubt that using `scope` with `StopWatch` is all that idiomatic, because stopping it is not the last thing you do with it. It's also designed with the idea that you can start and stop it multiple times (if you don't want to do that, arguably just using `MonoTime` directly makes more sense). So, while I'm sure that there are cases where using `scope` makes sense, I question that that's the typical use case. Regardless, the point of the docs is to show the basics of how to use it. I would hope that your typical D programmer could figure out when it's appropriate to use `scope` with `StopWatch` in their own code.